pkgs/by-name/im/imgpkg/package.nix

```diff
@@ -0,0 +1,36 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+let
+  pname = "imgpkg";
+  version = "0.42.1";
+
+  src = fetchFromGitHub {
+    owner = "carvel-dev";
+    repo = "imgpkg";
+    rev = "v${version}";
+    hash = "sha256-n8v27FpFpx0PBu/cCHgq128+fAjSGTEdLCG/g7MHCVg";
+  };
+in
+buildGoModule {
+  inherit pname version src;
```

Comment:
```suggestion
buildGoModule rec {
  pname = "imgpkg";
  version = "0.42.1";

  src = fetchFromGitHub {
    owner = "carvel-dev";
    repo = "imgpkg";
    rev = "v${version}";
    hash = "sha256-n8v27FpFpx0PBu/cCHgq128+fAjSGTEdLCG/g7MHCVg";
  };
```

Reply:
It was suggested here to avoid `rec` so now I'm unsure which one to go with 😄 

https://github.com/NixOS/nixpkgs/pull/336558#discussion_r1727435400

Reply:
`rec` is an antipattern, is it not?

Reply:
I can't see any info in the contribution guidelines for either avoiding or adhering to `rec` as a pattern. @SuperSandro2000 @SigmaSquadron can we agree on a style to approve? I'd quite like to get this merged so I can add some other missing packages

Reply:
In the interest of not nitpicking, I'd agree with either style. However, `rec` is [definitely on its way out](https://github.com/NixOS/nixpkgs/issues/315337), and shouldn't be recommended anymore. Ideally, one would use `finalAttrs` if the builder supports it, or `let...in` if not. As the official Nix documentation explicitly recommends against overusing `rec`, I believe that `let...in` or `finalAttrs` is the ideal solution here, but Sandro has far more review experience than me so I'll defer to his judgement.

Keep `rec`, for now. Once a greater consensus is reached, we can do a massive tree-wide to remove it.